### PR TITLE
Serverless: randomize times of periodic runs to prevent spikes

### PR DESCRIPTION
**Pull request by mgencur**
This only randomizes runs on OCP 4.9 and 4.8 which use cluster pools.

### CODE REVIEW

File: ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main__4.8.yaml [hunk 1]
```diff
@@ -164,7 +164,7 @@ tests:
     product: ocp
     timeout: 1h0m0s
     version: "4.8"
-  cron: 0 */12 * * *
+  cron: 0 2/14 * * *
```
Comment (cardil) [inline ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main__4.8.yaml:167]: Did you mean?

```suggestion
  cron: 0 2,14 * * *
```

File: ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main__4.9.yaml [hunk 2]
```diff
@@ -249,7 +249,7 @@ tests:
     product: ocp
     timeout: 1h0m0s
     version: "4.9"
-  cron: 0 */12 * * *
+  cron: 0 2/14 * * *
```
Comment (cardil) [inline ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main__4.9.yaml:252]: Did you mean?

```suggestion
  cron: 0 2,14 * * *
```

File: ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main-periodics.yaml [hunk 3]
```diff
@@ -482,7 +482,7 @@ periodics:
         secretName: result-aggregator
 - agent: kubernetes
   cluster: build03
-  cron: 0 */12 * * *
+  cron: 0 2/14 * * *
```
Comment (cardil) [inline ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main-periodics.yaml:485]: Did you mean?

```suggestion
  cron: 0 2,14 * * *
```

File: ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main-periodics.yaml [hunk 4]
```diff
@@ -865,7 +865,7 @@ periodics:
...
```
Comment (cardil) [inline ci-operator/jobs/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main-periodics.yaml:868]: Did you mean?

```suggestion
  cron: 0 2,14 * * *
```

**Review by cardil — CHANGES_REQUESTED** [review]
The `2/14` expression looks weird.

**Comment by cardil** [comment]
Supersedes https://github.com/openshift/release/pull/19843

File: ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main__4.8.yaml (hunk 1 above)
Comment (mgencur) [reply ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main__4.8.yaml:167]: Oops. Thanks for the catch.

**Review by rhuss — COMMENTED** [review]
Looks good to me. Do you know how long a typical run takes ? If its longer than 1 hour maybe a slightly larger spread might be even more helpful ?

**Comment by mgencur** [comment]
@rhuss On OCP 4.9 the cluster from a cluster-pool is used for so long before releasing:
* e2e - 1h44m
* e2e-mesh - 1h30m
* upgrades - 35m

On OCP 4.8 it takes slightly longer.

I could probably increase the time gap to 2 hours

**Comment by rhuss** [comment]
I think that would make sense to increase to 2 hours for the long-running tests and maybe 1hour for the upgrade test (you could do the upgrade test first and put one of the longer tests to the end to get results quicker for two tests, but that is probably irrelevant anyway).

**Comment by rhuss** [comment]
/approve
/lgtm

thanks !

**Pull request merged**